### PR TITLE
Warning fix

### PR DESCRIPTION
**Pull request by halfway258**
- added epochs (rounds) in source
- removed warnings
- fix "flask port in use" crash
- fixed missing data in python training loss graph

### CODE REVIEW

**Comment by halfway258** [comment]
verified better convergence of branch

**Review by leondavi — APPROVED** [review]

File: src_erl/Communication_Layer/http_Nerlserver/src/source/parser.erl [hunk 1]
```diff
@@ -65,10 +65,10 @@ parseCSV(SourceName, BatchSize, CSVData)->
 parse_file(SourceName, BatchSize,Data) ->
   % {ok, Data} = file:read_file(File_Address),
   Lines = re:split(Data, "\r|\n|\r\n", [{return,binary}] ),
-  CleanLines = [Line || Line <- Lines, Line /= []],
+  % CleanLines = [Line || Line <- Lines, Line /= []],
   ?LOG_INFO("split in data"),
   %% TODO: create NIF that does dataStrToNumericData. In = ["1,2,3,4", "5,6,7,8", ....] => out = [[1,2,3,4],[5,6,7,8],....] (make sure to take type into account)
-  ListOfLinesOfData = dataStrToNumericData(CleanLines),
+  ListOfLinesOfData = dataStrToNumericData(lists:droplast(Lines)),
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/source/parser.erl:71]: What is the meaning of this line? Please add a comment in code

**Review by leondavi — CHANGES_REQUESTED** [review]

File: src_py/apiServer/apiServer.py [hunk 2]
```diff
@@ -410,7 +411,7 @@ def accuracy_matrix(self, expNum):
             for j in range(labelsLen):
                 # print(f"worker {worker}, has {len(workerNeuronRes[worker][TRUE_LABLE_IND])} labels, with {len(workerNeuronRes[worker][TRUE_LABLE_IND][j])} samples")
                 # print(f"confusion {worker}:{j}, has is of {workerNeuronRes[worker][TRUE_LABLE_IND][j]}, {workerNeuronRes[worker][PRED_LABLE_IND][j]}")
-                confMatList[worker][j] = confusion_matrix(workerNeuronRes[worker][globe.TRUE_LABLE_IND][j], workerNeuronRes[worker][globe.PRED_LABLE_IND][j])
+                confMatList[worker][j] = confusion_matrix(workerNeuronRes[worker][globe.TRUE_LABLE_IND][j], workerNeuronRes[worker][globe.PRED_LABLE_IND][j], normalize='all')
```
Comment (leondavi) [inline src_py/apiServer/apiServer.py:414]: Make the normalization as an option.
 user might want to select in some more balanced cases

**Review by leondavi — CHANGES_REQUESTED** [review]

**Review by leondavi — APPROVED** [review]

**Comment by leondavi** [comment]
@halfway258 
You can merge it 🙂
